### PR TITLE
feat(meru AB): try assistant retractable drawer

### LINKED ISSUE #3778: [Meru - AB] Layout - Replace "Try" button by retractable preview
Details on what to do in the **[Figma](TODO)** (also, see how it works in the [Prototype](https://www.figma.com/proto/QOxHwppG64GtPocpDhS6Y7/Product?page-id=3366%3A31345&type=design&node-id=4900-40669&viewport=1173%2C2546%2C0.27&t=lV2M2VcNHLWO3Bpe-1&scaling=min-zoom&starting-point-node-id=4900%3A40669) + context & discussion in the [making issue](https://github.com/dust-tt/decisions/issues/164))

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

fixes https://github.com/dust-tt/dust/issues/3778

### Phone UI is unchanged:
builder_phone

### Preview works even with a "blank" new assistant
builder_new

### Convo is cleared & preview is reloaded when assitant changes
builder_update_instruct

It's hard to cover everything in the description, but this should be good enough for v1.
If naming isn't done yet, assistant is called "preview" in the try drawer

### CODE REVIEW

**Comment by philipperolet** [comment]
Weird scrollbar appearing. Might only be linux; but if it's windows too, we should fix

image

**Comment by philipperolet** [comment]
I notice the autoscroll is working fine :)

**Comment by philipperolet** [comment]
Nit: typing in instructions then opening preview => a reload occurs even if first time opening
Screencast from 2024-03-06 11-44-07

**Comment by fontanierh** [comment]
Thanks @philipperolet ! Trying to fix both these things now

**Review by philipperolet — APPROVED** [review]
LGTM :pray: Except for a handleclick that I'm unsure of (minor)
Thanks for contributing to :fish:

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 1]
```diff
@@ -217,6 +221,33 @@ export default function AssistantBuilder({
...
+    setPreviewDrawerOpenedAt(null);
+  };
+
+  const previewDrawerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    function handleClick(event: MouseEvent) {
+      const target = event.target;
+
+      if (!(target instanceof Node)) return;
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilder.tsx:241]: This is some useless leftover to a previous solution... should be removed thanks for catching !

[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilder.tsx]: deleted !

**Pull request merged**